src/env.js

```diff
@@ -0,0 +1,31 @@
+
+/**
+ * @see https://github.com/ipfs/ipfs-webui/issues/2078#issuecomment-1384444232
+ * @see https://github.com/ipfs/ipfs-webui/issues/2078#issuecomment-1384605253
+ * @returns {Promise<'local'|'kubo'|'webui.ipfs'>}
+ */
+
+const webuiRegex = /(?:webui-ipfs|webui\.ipfs)/
+const localhostRegex = /localhost/
+export async function getDeploymentEnv () {
+  const { origin } = globalThis.location
+  if (webuiRegex.test(origin)) {
+    if (localhostRegex.test(origin)) {
+      return 'local'
+    }
+    return 'webui.ipfs'
+  }
+  try {
+    const response = await fetch('/webui', { method: 'HEAD' })
+    if (response.redirected) {
```

Comment:
NIT: 
is this kubo check necessary? if not local or webui.ipfs.io it would be kubo correct?
is this check just a safeguard for future?

Reply:
Unfortunately, it is required as we can't really guarantee that it's kubo unless it meets the expectations of kubo. We should assume it's typical webui unless we can guarantee it's kubo. I added some tests to express this more clearly, but as lidel mentioned at https://github.com/ipfs/ipfs-webui/issues/2078#issuecomment-1384605253 it's susceptible to silent failures upon kubo's changing of behavior.

Reply:
that question was something i was wondering myself after addressing lidel’s latest comment. but i’d rather have some webui metrics for kubo under webui than non-kubo webui metrics listed for kubo

Reply:
makes sense, tests help too.  LGTM